testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jsonb/basic/JsonBindingDebugLoggingTest.java

```diff
@@ -0,0 +1,324 @@
+    * The ExceptionHandler processes it as a ReaderException.  The exception is
+    * written to the server.log by this class.  A Response object is created.  The
+    * exception message is assigned as the (String) entity in the response object.
+    * The response object is returned to the caller.
+    */
+   @Test
+   public void exceptionDuringServerReceive() throws Exception {
+
+      // count log messages before request
+      LogCounter errorStringLog = new LogCounter("ERROR", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      LogCounter resteasyExceptionLog = new LogCounter(".*DEBUG .* RESTEASY002305.*", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER, true);
+
+      LogCounter yassonExceptionLog = new LogCounter("Caused by: javax.json.bind.JsonbException", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter yassonStacktraceLog = new LogCounter("at org.eclipse.yasson", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter applicationExcpetionLog = new LogCounter(
+              "Caused by: java.lang.RuntimeException: "
+            + JsonBindingDebugLoggingItemCorruptedSet.class.getSimpleName(), true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      // perform request
+      WebTarget base = client.target(generateURL("/post"));
+      JsonBindingDebugLoggingItem wrongItem = new JsonBindingDebugLoggingItem();
+      wrongItem.setA(5);
+      Response response = base.request().post(Entity.entity(wrongItem,
+              MediaType.APPLICATION_JSON));
+
+      // check response
+      Assert.assertThat("Wrong response code", response.getStatus(), is(400));
+      Assert.assertThat("Response message doesn't contains expected message",
+              response.readEntity(String.class),
+              containsString("RESTEASY008200: JSON Binding deserialization error"));
+
+      // assert log messages after request in the server.log
+      Assert.assertThat("Application Exception should be logged",
+              applicationExcpetionLog.count(), is(1));
+      Assert.assertThat("RESTEasy exception should be logged",
+              resteasyExceptionLog.count(), is(1));
+      Assert.assertThat("Yasson exception should be logged",
+              yassonExceptionLog.count(), is(1));
+      Assert.assertThat("Yasson exception stacktrace should be logged",
+              yassonStacktraceLog.count(), greaterThan(0));
+
+      Assert.assertThat("There shouldn't be any error logs in server",
+              errorStringLog.count(), is(0));
+   }
+
+   /**
+    * @tpTestDetails Check exception during server receiving
+    * @tpSince RESTEasy 4.0.0.Beta7
+    *
+    * In this scenario the exception originates in JsonBindingProvider.readFrom.
+    * The exception is passed up the call stack to ClientResponse.readEntity which
+    * catches it as a RuntimeException.  The original exception is written to the
+    * server.log and returned to the caller.

```

Comment:
"server log"?

Reply:
Arquillian test class don't use @RunAsClient annotation, so test methods (with resteasy client invocation) is executed on server. This allows to print client debug logs to server.log and easily check it.

Reply:
Ah, I see. But in a normal application situation, the exception would be written on the client side, right? I think it might be worth mentioning since the nature of the logging is the point of the pull request.

Reply:
> But in a normal application situation, the exception would be written on the client side, right?

I depends what you mean by "normal application". I believe that "two servers, where first one server makes REST requests to the second one server" is common use case as well. Anyway, javadoc can be (of course) improved here ...

Reply:
re: "two servers, where first one server makes REST requests to the second one server""

That's true. We could just say "log" without specifying client or server.

Reply:
I added these test methods comments more for me as a (future) reminder guide of
the internal path each test was checking, because the path is complex.  I am
regretting doing it.  We are spending too much time on worrying over this single
word.  It's not like we are defining a spec rule here.  I have half a mind
to yank all my method comments.   Can we move on?